infra/blockchain/evm.go

```diff
@@ -18,14 +19,19 @@ type EvmHandler struct {
 }
 
 var clients map[int64]EvmHandler
+var clientsMapLock sync.Mutex
 
+// Call it at the application entry point
 func SetupEvmEnv() {
 	clients = make(map[int64]EvmHandler)
 }
 func GetEvmHandler(chainId int64) EvmHandler {
 	return clients[chainId]
 }
 func AddChainClient(chain models.Chain) {
+	clientsMapLock.Lock()
+	defer clientsMapLock.Unlock()
```

Comment:
if use mutex lock, should also lock/unlock before read to avoid data race condition.